core/arch/arm32/kernel/thread.c

```diff
@@ -73,6 +148,61 @@ thread_pm_handler_t thread_system_reset_handler_ptr;
 
 static unsigned int thread_global_lock = UNLOCK;
 
+#ifdef CFG_WITH_STACK_CANARIES
```

Comment:
Not a big deal, but what about moving the ifdef from
```
#ifdef CFG_WITH_STACK_CANARIES
static void init_canaries(void)
{ ...
#else 
static void init_canaries(void)
{...
#endif
```
to instead 
```
static void init_canaries(void)
{
#ifdef CFG_WITH_STACK_CANARIES
#endif
}
void thread_check_canaries(void)
{
#ifdef CFG_WITH_STACK_CANARIES
#endif
}
```
I found this in other places in a couple of locations in this patch.